.github/workflows/pr-check-pipfile.yml

```
@@ -0,0 +1,140 @@
+name: pr-check-pipfile
+
+# pull_requestで何かあった時に起動する
+on:
+  pull_request:
+    paths:
+      - "Pipfile*"
+
+env:
+  WORKON_HOME: /tmp/.venv
+
+jobs:
+  # Pipfileに差分があれば、Pipfile.lockを更新する
+  pr-check-pipfile:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8, 3.7]
+      fail-fast: false
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2.1.2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install pipenv
+        run: curl https://raw.githubusercontent.com/pypa/pipenv/master/get-pipenv.py | python
+      - name: pipenv lock
+        run: pipenv lock
+      - name: pipfile mv
+        run: mv Pipfile.lock Pipfile.lock-${{ matrix.python-version }}
+      # 差分があったときは差分を出力する
+      - name: Show diff
+        id: diff
+        run: |
+          git diff
+      # 差分があったときは、コミットを作りpushする
+      - name: Push
+        if: github.event.pull_request.head.repo.full_name == github.repository && steps.diff.outcome != ''
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        run: |
+          git config user.name "hatohakaraage"
+          git config user.email "hatohakaraage@example.com"
+          git add -u
+          git commit -m "鳩は唐揚げ！(自動で直してあげたよ！)"
+          git push -f https://${{github.actor}}:${{secrets.GITHUB_TOKEN}}@github.com/${{github.repository}}.git HEAD:refs/heads/pipfile-${PYTHON_VERSION//\./_}-${{github.event.pull_request.head.ref}}
+      # pushしたブランチでPRを作る
+      - name: Create PullRequest
+        uses: actions/github-script@v3
+        if: github.event.pull_request.head.repo.full_name == github.repository && steps.diff.outcome != ''
+        env:
+          PYTHON_VERSION: ${{ matrix.python-version }}
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const common_params = {
+              owner: context.repo.owner,
+              repo: context.repo.repo
+            }
+            const pull_params = {
+              head: "refs/heads/pipfile-" + process.env.PYTHON_VERSION.replace('.', '_') + "-${{github.event.pull_request.head.ref}}",
```

```suggestion
              head: "${{github.event.pull_request.head.repo.owner.login}}:pipfile-" + process.env.PYTHON_VERSION.replace('.', '_') + "-${{github.event.pull_request.head.ref}}",
```

このように指定するとうまく対象を絞れる
ref: https://developer.github.com/v3/pulls/#list-pull-requests

---

`${{github.event.pull_request.head.repo.owner.login}}` はpushの場合取得できないので、固定値 ( `dev-hato` ) の方が良いかも